### DISCUSSION TITLE
Contributor Tasks: Accessibility and Testing

### DISCUSSION THREAD

**Discussion by nwanduka** [camicroscope/caMicroscope#836] [Q&A] [answered]
Hi everyone, hope we're all enjoying contributing to the project.

I'm working on the Accessibility task in the contributor task list. It struck me that there is an overlap between accessibility and testing. So rather than just identify accessibility bugs, we could set up Github workflows to test for accessibility whenever a change is made to the repository. This way, we can catch accessibility bugs quicker and fix them.

I've spent a chunk of my time reading about the WCAG 2.1 standards, how to check for accessibility using tools like Axe and Google Lighthouse. I've made some progress in that regard and succeeded in identifying some accessibility issues within caMicroscope. Where I'm facing a challenge is figuring out how to set up Github workflows that will automatically test for these issues. I've devoured a substantial amount of Github documentation on the subject but it appears it was written with developers in mind. As a UX designer, most of the explanation flies over my head. 

I'm hoping to collaborate with other community members who are knowledgeable in the field and may already be contributing to testing. Please, if this is an area you're interested in, I'd appreciate your response to this message.

@birm

**Comment by We13b-MD** [comment]
Yes sure I will be happy to assist

**Reply by nwanduka** [reply]
Awesome. Thank you @We13b-MD. I and some other contributors have raised a couple of assessibility issues. The common ones are:
1. [Low color contrast](https://github.com/camicroscope/caMicroscope/issues/829)
2. [Zooming is disabled in mobile view](https://github.com/camicroscope/caMicroscope/issues/830)
3.  [html element doesn't have the lang attribute and some others.](https://github.com/camicroscope/caMicroscope/issues/837)

Do you think it's something you can take on; set up a workflow that will automatically test for these issues when a push is made to the repository?

**Reply by We13b-MD** [reply]
Yeah sure can you send a screenshot of the issue regarding low color contrast

**Reply by nwanduka** [reply, marked as answer]
![image](https://github.com/camicroscope/caMicroscope/assets/122698422/71935662-aaac-451c-9884-c6a58c260760)

![low color contrast](https://github.com/camicroscope/caMicroscope/assets/122698422/cd69d2a6-1f55-4f6d-845f-acb9b901c6e0)

**Reply by nwanduka** [reply]
Does this help?

**Reply by We13b-MD** [reply]
Yes let me take a good look at it will get back to you I will try setting up the workflow if I have any issues I will let you know

**Reply by nwanduka** [reply]
Alright. No problem. When you're done, I'd love to know how you set up the workflow.

**Reply by nwanduka** [reply]
I think you may find this helpful: https://github.com/dequelabs/axe-core

**Reply by birm** [reply]
@nwanduka this is exactly the sort of thing at least I was thinking of here. Lovely work :)

**Reply by We13b-MD** [reply]
@nwanduka currently working on the axe-core library for testing had to install wsl extension I guess the ace-core library uses Linux is and not windows if you have further suggestions let me know

**Reply by gbengaoluwadahunsi** [reply]
Awesome @nwanduka

**Reply by We13b-MD** [reply]
@nwanduka @birm i was able to integrate the axe-core library after reading its documention i tried running the tests with one of the elements take a look

https://github.com/camicroscope/caMicroscope/assets/122805291/8b9e4787-d6a3-4210-8cd9-b2861f9b8939

here is the code sample : 
![codesampleaxe](https://github.com/camicroscope/caMicroscope/assets/122805291/9b3358dd-07d2-43d0-bdb6-eea6ebcc8748)


i tested the select element in the table.html file  you can see everything in the video i hope this helps. If there are any additions let me know

**Reply by nwanduka** [reply]
Is the video showing the test results? I'm unsure what "element" you tested. You mentioned the table.html file. Please in which folder is that located?

**Reply by nwanduka** [reply]
By the way, thank you so much for working on this. It's one step forward from where we started 👏🏽

**Reply by We13b-MD** [reply]
I tested the select element on the slides section it does not have a select name or label

**Reply by We13b-MD** [reply]
The table.html file is in the apps section

**Reply by sarahannie** [reply]
@We13b-MD looking at your code sample and reading the documentation for axe , can you run me through how you were able to test for accessibility issues on table.html

**Reply by We13b-MD** [reply]
@sarahannie Go to https://github.com/dequelabs/axe-core
This should teach you how to install axe in your visual studio code 
Go to axe documentation just type ace-core documentation on your browser 
In the axe documentation you will see how to get started  testing for accessibility in your code 
Get the axe cdn script tag and include it in the file you want to work on.
All there should get yin started with axe-core library for testing.
Good luck testing

**Comment by blessdamy** [comment]
Hello everyone. 
Please, I need help, I keep getting this error after running both docker-compose -f caMicroscope.yml build and docker-compose -f caMicroscope.yml up
![CaM](https://github.com/camicroscope/caMicroscope/assets/111387281/7018b902-8941-4b36-8f02-1d1dc5174381)

**Reply by We13b-MD** [reply]
When building again add no cache to the build command

**Reply by blessdamy** [reply]
Thank you @We13b-MD. I will implement that
